Test flakes: scp broken pipe

`scp` outputs to stderr resulting in `EPIPE`:

<details><summary><code>Logs</code></summary>

```
--- FAIL: TestInvalidDir (0.09s)
    --- FAIL: TestInvalidDir/parent_directory (0.05s)
        scp_test.go:275: 
            	Error Trace:	scp_test.go:275
            	Error:      	Expect "<nil>" to match ".*invalid name.*"
            	Test:       	TestInvalidDir/parent_directory
FAIL
coverage: 67.1% of statements
FAIL	github.com/gravitational/teleport/lib/sshutils/scp	0.265s
```
</details>

--- a/lib/sshutils/scp/scp.go
+++ b/lib/sshutils/scp/scp.go
@@ -403,10 +403,10 @@ func (cmd *command) serveSink(ch io.ReadWriter) error {
 	}
 	var st state
 	st.path = localDir
-	var b = make([]byte, 1)
+	var b [1]byte
 	scanner := bufio.NewScanner(ch)
 	for {
-		n, err := ch.Read(b)
+		n, err := ch.Read(b[:])
 		if err != nil {
 			if err == io.EOF {
 				return nil

--- a/lib/sshutils/scp/scp_test.go
+++ b/lib/sshutils/scp/scp_test.go
@@ -265,14 +265,27 @@ func TestInvalidDir(t *testing.T) {
 	for _, tt := range testCases {
 		tt := tt
 		t.Run(tt.desc, func(t *testing.T) {
-			scp, in, out, _ := newCmd("scp", "-v", "-r", "-f", tt.inDirName)
-			rw := &readWriter{out, in}
+			t.Parallel()
+
+			scp, stdin, stdout, stderr := newCmd("scp", "-v", "-r", "-f", tt.inDirName)
+			rw := &readWriter{r: stdout, w: stdin}
+
+			doneC := make(chan struct{})
+			// Service stderr
+			go func() {
+				io.Copy(ioutil.Discard, stderr)
+				close(doneC)
+			}()
 
 			err := scp.Start()
 			require.NoError(t, err)
 
 			err = cmd.Execute(rw)
 			require.Regexp(t, tt.err, err)
+
+			stdin.Close()
+			<-doneC
+			scp.Wait()
 		})
 	}
 }
@@ -365,26 +378,26 @@ func TestSCPParsing(t *testing.T) {
 }
 
 func runSCP(cmd Command, args ...string) error {
-	scp, in, out, _ := newCmd("scp", args...)
-	rw := &readWriter{out, in}
+	scp, stdin, stdout, _ := newCmd("scp", args...)
+	rw := &readWriter{r: stdout, w: stdin}
 
 	errCh := make(chan error, 1)
 
 	go func() {
+		defer close(errCh)
 		if err := scp.Start(); err != nil {
-			errCh <- trace.Wrap(err)
+			errCh <- err
 			return
 		}
 		if err := cmd.Execute(rw); err != nil {
-			errCh <- trace.Wrap(err)
+			errCh <- err
 			return
 		}
-		in.Close()
+		stdin.Close()
 		if err := scp.Wait(); err != nil {
-			errCh <- trace.Wrap(err)
+			errCh <- err
 			return
 		}
-		close(errCh)
 	}()
 
 	select {

